### PR TITLE
requests: support a list of requests and no-session support

### PR DESCRIPTION
**Pull request by twilfredo**

### CODE REVIEW

**Comment by twilfredo** [comment]
This is a modified approach from https://github.com/westerndigitalcorporation/spdm-utils/pull/97 to address the review comments.

File: src/main.rs [hunk 1]
```diff
@@ -264,6 +324,207 @@ pub enum RequestCode {
...
+                .parse::<RequestCode>()
+                .map_err(|e| format!("Failed to parse request: {e}"))?;
+            // Track the index after the ','
+            prev_req_start_at = i + 1;
+            purse.push(request_parsed.clone());
+        }
+    }
+
+    Ok(purse)
+}
```
Comment (twilfredo) [inline src/main.rs:531]: Any thoughts on if this could be made better? Currently just manually parsing...

File: src/main.rs [hunk 2]
```diff
@@ -264,6 +329,207 @@ pub enum RequestCode {
     },
 }
 
+impl std::str::FromStr for RequestCode {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        // Support only the requests that can either function with default
+        // options, or does not require any arguments.
+        warn!("Input {s}");
```
Comment (alistair23) [inline src/main.rs:338]: `info!` instead of `warn!`?

File: src/main.rs [hunk 3]
```diff
@@ -88,15 +88,80 @@ struct Args {
     /// option is not specified.
     #[arg(long, default_value_t = 2323)]
     qemu_port: u16,
+
+    /// Do not initialize an SPDM session with the responder, instead, only
+    /// send the requests specified
+    #[arg(long)]
+    no_session: bool,
```
Comment (alistair23) [inline src/main.rs:95]: Default true maybe just to be explicit?

File: src/main.rs (hunk 1 above)
Comment (alistair23) [reply src/main.rs:531]: Not that I know of. I didn't see anything obvious in clap.

I think we should add some unit tests though that call `parse_request_codes()` with a range of tests that are run as part of `cargo test`

**Review by alistair23 — CHANGES_REQUESTED** [review]
Looks great! Let's add some unit tests though

File: src/main.rs (hunk 1 above)
Comment (twilfredo) [reply src/main.rs:531]: I was thinking that too, I'll add some

File: src/main.rs (hunk 2 above)
Comment (twilfredo) [reply src/main.rs:338]: that's actually left over from testing, I'll remove it!

File: src/main.rs (hunk 3 above)
Comment (twilfredo) [reply src/main.rs:95]: good point!

**Comment by twilfredo** [comment]
> Looks great! Let's add some unit tests though

Added a two tests cases that cover a range of positive and negative tests. Let me know if the coverage is ok

**Review by alistair23 — APPROVED** [review]

**Pull request merged**